tests/jax/models/MNIST_CNN/CNN.py

```diff
@@ -0,0 +1,29 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import jax
+from flax import linen as nn
+
+class CNN(nn.Module):
+    @nn.compact
+    def __call__(self, x, *, train: bool):
```

Comment:
This `*` might be completely unneeded here (not sure if inheriting `nn.Module` or `@nn.compact` decorator force it though), but leave it as is since it provides a perfect example of a tester which uses every part of override-interface.